File: code/design-patterns/behavioral/visitor/python/visitor_usage.py
from visitor import Circle, Square, Triangle, AreaCalculator, PerimeterCalculator

shapes = [Circle(10), Square(5), Triangle(6, 8, 8)]
area_calculator = AreaCalculator()
perimeter_calculator = PerimeterCalculator()

for shape in shapes:
    shape.accept(area_calculator)
    shape.accept(perimeter_calculator)

File: code/design-patterns/behavioral/visitor/go/visitor_usage.go
package main

func main() {
	shapes := []Shape{
		&Circle{Radius: 10},
		&Square{Side: 5},
		NewTriangle(6, 8, 8),
	}

	areaCalculator := &AreaCalculator{}
	perimeterCalculator := &PerimeterCalculator{}

	for _, shape := range shapes {
		shape.Accept(areaCalculator)
		shape.Accept(perimeterCalculator)
	}
}
